File: test/python/widget/get_auth_2_test.py
import os
import pytest
import asyncio
from dana.utils.snap_configuration import SnapConfiguration, AuthSettings, Env
from dana.ipg.v1.api import IPGApi
from dana.api_client import ApiClient
from helper.util import with_delay
from automate_oauth import automate_oauth

configuration = SnapConfiguration(
    api_key=AuthSettings(
        PRIVATE_KEY=os.environ.get("PRIVATE_KEY"),
        ORIGIN=os.environ.get("ORIGIN"),
        X_PARTNER_ID=os.environ.get("X_PARTNER_ID"),
        ENV=Env.SANDBOX
    )
)

with ApiClient(configuration) as api_client:
    api_instance = IPGApi(api_client)

@pytest.fixture(scope="module")
def test_get_auth_reference_number():
    from uuid import uuid4
    return str(uuid4())

@with_delay()
def test_get_auth_success():
    output = asyncio.run(automate_oauth())
    if not output:
        pytest.fail("automate_oauth() did not return any value.")
    auth_code = output
    if not auth_code:
        pytest.fail("Auth Code was not returned by automate_oauth().")

File: test/php/widget/GetAuth2Test.php
<?php

namespace DanaUat\Widget;

use PHPUnit\Framework\TestCase;
use Dana\Widget\v1\Api\WidgetApi;
use Dana\Configuration;
use Dana\ObjectSerializer;
use Dana\Env;
use DanaUat\Helper\OauthUtil;
use DanaUat\Helper\Util;

class GetAuth2Test extends TestCase
{
    private static $titleCase = 'GetAuth2';
    private static $jsonPathFile = 'resource/request/components/Widget.json';
    private static $apiInstance;
    private static $seamlessData;
    private static $seamlessSign;
    private static $phoneNumber = '083811223355';

    public static function setUpBeforeClass(): void
    {
        $configuration = new Configuration();
        $configuration->setApiKey('PRIVATE_KEY', getenv('PRIVATE_KEY'));
        $configuration->setApiKey('ORIGIN', getenv('ORIGIN'));
        $configuration->setApiKey('X_PARTNER_ID', getenv('X_PARTNER_ID'));
        $configuration->setApiKey('ENV', Env::SANDBOX);
        self::$apiInstance = new WidgetApi(null, $configuration);
    }

    /**
     * Should give success response for get auth2 scenario
     */
    public function testGetAuth2Success(): void
    {
        $this->markTestSkipped('Skipping testGetAuth2Success ');
        // Util::withDelay(function () {
        //     $queryParams = [
        //         'partnerId' => getenv('X_PARTNER_ID'),
        //         'timestamp' => '2024-08-31T22:27:48+00:00',
        //         'externalId' => 'test',
        //         'channelId' => getenv('X_PARTNER_ID'),
        //         'scopes' => 'CASHIER,AGREEMENT_PAY,QUERY_BALANCE,DEFAULT_BASIC_PROFILE,MINI_DANA',
        //         'redirectUrl' => 'https://google.com',
        //         'state' => OauthUtil::generateUUID(),
        //         'isSnapBI' => 'true',
        //         'seamlessData' => self::$seamlessData,
        //         'seamlessSign' => self::$seamlessSign
        //     ];
        //     // Get the OAuth URL
        //     $response = $this->getOauthUrl($queryParams);

        //     $this->assertEquals('200', $response['statusCode'], 'Expected success response code');
        // });
    }
}
